### PR TITLE
use Telescope-based cosmic-proto

### PR DESCRIPTION
**Pull request by turadg**
0.4.0 builds with Telescope, which provides some client support we were hand rolling

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1173,6 +1216,18 @@
     cosmjs-types "^0.8.0"
     long "^4.0.0"
 
+"@cosmjs/proto-signing@^0.32.2":
```
Comment (0xpatrickdev) [inline yarn.lock:1219]: nit: it seems like we now have `0.30.x` and `0.32.x` versions for many of the `@cosmjs/*` deps

**Review by 0xpatrickdev — APPROVED** [review]

File: yarn.lock (hunk 1 above)
Comment (turadg) [reply yarn.lock:1219]: Good to raise. It looks to be mostly due to leapwallet pinning versions. We could use a resolution to override but that can be brittle.

I'll wait for @samsiegart 's input before merging. Also WRT test plan

File: src/vite-env.d.ts [hunk 2]
```diff
@@ -45,8 +45,3 @@ declare module '@agoric/inter-protocol/src/vaultFactory/math' {
 declare module 'react-view-slider' {
   export const ViewSlider;
 }
-
-declare module '@agoric/cosmic-proto/swingset/query.js' {
```
Comment (samsiegart) [inline src/vite-env.d.ts:49]: Woohoo!

File: src/components/ProvisionSmartWalletNoticeDialog.tsx [hunk 3]
```diff
@@ -12,14 +12,19 @@ const useSmartWalletFeeQuery = (rpc: string | null) => {
     const fetchParams = async () => {
       assert(rpc);
       try {
-        const params = await querySwingsetParams(rpc);
+        const client = await agoric.ClientFactory.createRPCQueryClient({
```
Comment (samsiegart) [inline src/components/ProvisionSmartWalletNoticeDialog.tsx:15]: Remark: the wallet connection also makes its own query client, ideally at some point we could share between these two places somehow. I think the only main downside currently is an extra "status" query when you create an extra client.

**Review by samsiegart — APPROVED** [review]
LGTM, code is much cleaner and thanks for the descriptive commit messages

File: yarn.lock (hunk 1 above)
Comment (samsiegart) [reply yarn.lock:1219]: I tested the deployed preview and looks like it worked. I don't know much about the version thing, it's good to be aware of though, maybe upgrading leap elements would fix it.

**Pull request merged**